### ISSUE TITLE
Does HYPRE support multiple instances at the same time?

### ISSUE THREAD

**Issue by stgeke** [hypre-space/hypre#775]

**Comment by liruipeng** [comment]
@stgeke No, it doesn't, if I understand the question. Are you asking about using hypre in a threaded region, so each thread has an instance?

**Comment by stgeke** [comment]
No I want to use HYPRE to solve two different system but not at the same time. For each I want to use different settings.

**Comment by liruipeng** [comment]
I may miss the gist of this issue. If not trying to solve 2 systems at the same time, why do you need 2 instances of hypre? About different settings, do you mean something like different AMG parameters? Thanks!

**Comment by stgeke** [comment]
Yes the two will use different AMG parameters. I just want to double if I would run into any issues. My current understanding is that I just need to setup the two "instances" and store the returned matrix + vectors. Is that correct?

**Comment by liruipeng** [comment]
@stgeke Sorry for replying your last message late. If you don't need to solve them at the same time, can you just do setup and solve for the 1st system, and set the new parameters, setup and solve the 2nd one? You need to store the matrices and vectors for all the systems somehow, but we shouldn't need to have 2 "instances" of hypre, I guess. You were right in your email, there is a global `_hypre_handle` data struct along with other global variables at places, so we don't generally support multiple instances of hypre. Does this make sense? Thanks!

**Comment by stgeke** [comment]
In my time stepping code the matrix doesn't change. Ideally I want to call setup only once but run the solver at every timestep. Is this also supported? Probably not because there are some globals?

**Comment by liruipeng** [comment]
@stgeke I don't see issues with "setup only once but run the solver at every time step". We have backward Euler time stepping test. You can setup AMG once and call solve at each time with a new vector (without destroying the matrix). I don't see problems with the globals. (You don't need to call `HYPRE_Init` multiple times where basically the globals are created).

**Comment by victorapm** [comment]
@stgeke Has this issue been resolved? Can we close it? Thanks!
